Userland/Libraries/LibC/fcntl.cpp

```diff
@@ -104,6 +104,13 @@ int posix_fadvise(int fd, off_t offset, off_t len, int advice)
     return 0;
 }
 
+// https://pubs.opengroup.org/onlinepubs/007904975/functions/posix_fallocate.html
+int posix_fallocate(int fd, off_t offset, off_t len)
+{
+    // posix_fallocate does not set errno
+    return static_cast<int>(syscall(SC_posix_fallocate, fd, &offset, &len));
```

Comment:
Doesn't this return negative error numbers? Is that expected?

Reply:
It states:
> Upon successful completion, posix_fallocate() shall return zero; otherwise, an error number shall be returned to indicate the error.
So idk

Reply:
Judging by the implementations in musl and glibc, as well as various snippets that are floating around the internet, the return values are indeed meant to be positive error numbers.